Add SpawnFeature and register in Entrypoint

DinoMarlir opened the pull request:
Introduces the SpawnFeature class, which provides a /spawn command to teleport players to the world's spawn location. The feature is registered in Entrypoint to enable its functionality.

DinoMarlir committed:
Add SpawnFeature and register in Entrypoint

Introduces the SpawnFeature class, which provides a /spawn command to teleport players to the world's spawn location. The feature is registered in Entrypoint to enable its functionality.

DinoMarlir committed:
remove todo

DinoMarlir committed:
Update exception message in getSpawnLocation

Changed the exception message in getSpawnLocation from German to English for consistency and clarity.

DinoMarlir committed:
Restrict /spawn command to players only

Added a command requirement so only Player entities can execute the /spawn command. Simplified command execution logic by removing sender type checks.

FogknightMC commented:
Der Spawn-Command sollte auf Folia problemlos funktionieren, weil Folia Events & Commands immer im korrekten Executor-Threadcontext aufgerufen werden. Da du hier nicht mit weiteren Spielern interagierst brauchst du dir also keine Sorgen zu machen. 

Die Verwendung von Player.teleportAsync() ist wichtig, da Player.teleport() auf Folia unter keinen Umständen funktioniert.  

Der World-Check ist denke ich überflüssig. Würden wir in der plugin.yml einfach "load" auf POSTWORLD setzen, gäbe es niemals eine null-value hier, da die Welt immer zuerst erstellt oder geladen wird.

DinoMarlir committed:
load plugin post world + removed null check in SpawnFeature

DinoMarlir commented:
> Der Spawn-Command sollte auf Folia problemlos funktionieren, weil Folia Events & Commands immer im korrekten Executor-Threadcontext aufgerufen werden. Da du hier nicht mit weiteren Spielern interagierst brauchst du dir also keine Sorgen zu machen.
> 
> Die Verwendung von Player.teleportAsync() ist wichtig, da Player.teleport() auf Folia unter keinen Umständen funktioniert.
> 
> Der World-Check ist denke ich überflüssig. Würden wir in der plugin.yml einfach "load" auf POSTWORLD setzen, gäbe es niemals eine null-value hier, da die Welt immer zuerst erstellt oder geladen wird.

Danke für die Anmerkungen, ich habe das direkt mal eingebaut.

DinoMarlir committed:
Refactor disable() to use expression body

Changed the override of disable() in SpawnFeature to use an expression body returning Unit for conciseness.

DinoMarlir committed:
cleaned up imports
